### ISSUE TITLE
Add an `envoylint` tool

### ISSUE THREAD

**Issue by phlax** [envoyproxy/envoy#13233]
If we create jsonschema for validating envoy yaml/json config files (see #13078 ) we can re-use the schema for a `cli` linter - which could eg be used to validate snippets in docs etc (see #8837 and #13231)

im thinking this could also have wider use - for deployments or eg in gitops CI for end users

*Description*:
>Describe the the desired behavior, what scenario it enables and how it
would be used.

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by mattklein123** [comment]
@phlax see prior art here in our various config tooling: https://github.com/envoyproxy/envoy/tree/master/test/tools and https://www.envoyproxy.io/docs/envoy/latest/install/tools/tools. See also related tickets here: https://github.com/envoyproxy/envoy/issues?q=is%3Aopen+is%3Aissue+label%3Aarea%2Froute_check_tool. 

It would be great to unify all of these efforts where possible.

**Comment by phlax** [comment]
> It would be great to unify all of these efforts where possible.

yep

the config load tool does a bit more than static analysis so i think we would want to at least retain that functionality

**Comment by phlax** [comment]
after a little  more dev/research on this im wondering whether its possible/better to do this directly from protobuf - ie without the jsonschema step.

either way it would be good to publish the jsonschema to https://www.schemastore.org/json/ so existing jsonschema tooling beyond vscode can make use of it

**Comment by phlax** [comment]
see https://envoylint.com/

**Comment by ysawa0** [comment]
Hey @phlax, didn't realize you posted it here. I updated the site to now run against Envoy binaries with three different versions supported (1.16, 1.14, 1.12).

**Comment by phlax** [comment]
> Hey @phlax, didn't realize you posted it here.

yep, i still think a static analysis tool could be useful, and was collating related info here

> I updated the site to now run against Envoy binaries with three different versions supported (1.16, 1.14, 1.12).

cool, ive pointed a couple people towards it already
